scripts/terminology_valueset_sync.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python
+
+"""This script provides the ability to pull down various medical terminology
+valusets required for processing eICR data, specifically adding codes to
+different data elements within an eICR message, such as Lab Order Name or
+Lab Result Interpretation.
+
+Current Available Valuesets:
+    - Lab Names (Ordering & Resulting) - LOINC
+    - Lab Orders - LOINC
+    - Lab Observations - LOINC
+    - Lab Result Value - SNOMED
+    - Lab Result Interpretation - HL7 Observation Interpretations
+
+Requirements:
+    - SNOMED - requires an UMLS API KEY stored in an environment variable:
+        - UMLS_API_KEY
+    - LOINC - requires a LOINC username and password stored in environment variables:
+        - LOINC_USERNAME
+        - LOINC_PWD
+"""
+
+import argparse
+import csv
+import os
+import sys
+
+import requests
+
+# Set Terminology URLS
+LOINC_BASE_URL = "https://loinc.regenstrief.org/searchapi/loincs?"
+LOINC_LAB_ORDER_SUFFIX = "query=orderobs:Order+OR+orderobs:Both&rows=500"
+LOINC_LAB_RESULT_SUFFIX = "query=orderobs:Observation+OR+orderobs:Both&rows=500"
+LOINC_LAB_NAMES_SUFFIX = "query=orderobs:Order+OR+orderobs:Both+OR+orderobs:Observation"
+HL7_LAB_INTERP_URL = (
+    "https://www.fhir.org/guides/stats2/valueset-us.nlm.vsac-2.16.840.1.113883.1.11.78.json"
+)
+UMLS_SNOMED_LAB_VALUES_URL = (
+    "https://uts-ws.nlm.nih.gov/rest/content/current/source/SNOMEDCT_US/260245000/descendants"
+)
+
+# Get Terminology Usernames and Passwords
+LOINC_USERNAME = os.environ.get("LOINC_USERNAME")
+LOINC_PWD = os.environ.get("LOINC_PWD")
+UMLS_API_KEY = os.environ.get("UMLS_API_KEY")
+
+# CSV file settings
+CSV_DIRECTORY = "tmp/"
+
+
+def get_umls_snomed_lab_values():  # noqa: D103
+    if UMLS_API_KEY is None:
+        raise KeyError("UMLS_API_KEY Environment Variable must be set to a proper UMLS API Key!")
+    snomed_filename = "snomed_lab_value.csv"
+    page_num = 1
+    params = {"apiKey": UMLS_API_KEY, "pageNumber": page_num, "pageSize": 500}
+    umls_response = requests.get(UMLS_SNOMED_LAB_VALUES_URL, params=params)
+    snomed_row_count = 0
+    snomed_rows = []
+
+    while umls_response.status_code == 200:
+        # NOTE: the UMLS responses are a bit slow
+        #  you can use the print statement below to get a
+        #  better idea of the progress if needed.
+        # print(f"Processing SNOMED page {page_num}")
+        umls_results = umls_response.json().get("result")
+
+        for result in umls_results:
+            snomed_code = result.get("ui")
+            snomed_text = result.get("name")
+            if snomed_code and snomed_text:
+                result_row = {"code": snomed_code, "text": snomed_text}
+                snomed_rows.append(result_row)
+                snomed_row_count += 1
+
+        page_num += 1
+        params = {"apiKey": UMLS_API_KEY, "pageNumber": page_num}
```

Comment:
You're missing the `pageSize` here, alternatively you could also do this
```suggestion
        params['pageNumber'] += 1
```

Reply:
Ah, good catch, which is weird that you would have to specify the pagesize each time, but I"ll add it here.  Thanks!!